src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1022,9 +1054,16 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             // TryComplete, we may miss diagnostics.
                             var hostDiagnostics = analyzerDriver.GetDiagnosticsAsync(compilation).Result;
                             diagnostics.AddRange(hostDiagnostics);
-                            if (hostDiagnostics.Any(IsReportedError))
+
+                            if (!diagnostics.IsEmptyWithoutResolution)
                             {
-                                success = false;
+                                // Apply diagnostic suppressions for analyzer and/or compiler diagnostics from diagnostic suppressors.
+                                analyzerDriver.ApplyProgrammaticSuppressions(diagnostics, compilation);
+
+                                if (diagnostics.AsEnumerable().Any(IsReportedError))
```

Comment:
>AsEnumerable() [](start = 48, length = 14)

Is addition of ```AssEnumerable``` necessary?

Reply:
`DiagnosticBag` doesn't implement `IEnumerable<T>`. Is there a better way to do this?

Reply:
> DiagnosticBag doesn't implement IEnumerable<T>. Is there a better way to do this?

If there is no other way to do this today, consider if it would make sense to add an API to ```DiagnosticBag``` similar to ```HasAnyErrors```.

---
In reply to: [292202350](https://github.com/dotnet/roslyn/pull/36067#discussion_r292202350) [](ancestors = 292202350)

Reply:
I ended up using the existing helper HasUnsuppressedErrors in this file, which does the same check. This will also mean we have one lesser place to handle when we fix https://github.com/dotnet/roslyn/issues/36215